datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -1004,9 +1004,13 @@ impl GroupedHashAggregateStream {
     /// Updates skip aggregation probe state.
     fn update_skip_aggregation_probe(&mut self, input_rows: usize) {
         if let Some(probe) = self.skip_aggregation_probe.as_mut() {
-            // Skip aggregation probe is not supported if stream has any spills,
-            // currently spilling is not supported for Partial aggregation
-            assert!(self.spill_state.spills.is_empty());
+            // Skip aggregation probe is only supported in Partial aggregation.
+            // And it is not supported if stream has any spills even in Partial aggregation.
+            // Although currently spilling is actually not supported in Partial aggregation,
+            // it is possible to be supported in future, so we also add an assertion for it.
+            assert!(
+                self.mode == AggregateMode::Partial && self.spill_state.spills.is_empty()
```

Comment:
In partial mode, `skip_aggregation_probe` should be None

Reply:
Yes, these are two defensive assertions.

Reply:
I want to make that update_skip_aggregation_probe will be only called in partial more clear by someway.

Maybe we can just let the update_skip_aggregation_probe will only be called in partial?

if self.mode == AggregateMode::Partial {
    self.update_skip_aggregation_probe(input_rows);
}

And we make self.skip_aggregation_probe.as_mut().unwarp in update_skip_aggregation_probe?

Reply:
It looks good to me, and also we could do the same for other skip aggregation logic like `switch_to_skip_aggregation` and `should_skip_aggregation`

Reply:
We could even have pattern matching on mode, trade duplicated function call with more clear code path for partial and non-partial mode

```rust

match self.mode {
AggregateMode::Partial => {}
_ => {} 
}
```

Reply:
> It looks good to me, and also we could do the same for other partial mode only like `switch_to_skip_aggregation`, `should_skip_aggregation` and `emit_early_if_necessary`

It is really a good suggestion for readability.
I have tried to refactor the codes following the suggestion, I thikn it indeed clearer for me.